lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -0,0 +1,472 @@
+}
+
+type defaultEnrollEKSClustersClient struct {
+	*eks.Client
+	stsClient *sts.Client
+}
+
+func (d *defaultEnrollEKSClustersClient) GetCallerIdentity(ctx context.Context, params *sts.GetCallerIdentityInput, optFns ...func(*sts.Options)) (*sts.GetCallerIdentityOutput, error) {
+	return d.stsClient.GetCallerIdentity(ctx, params, optFns...)
+}
+
+func (d *defaultEnrollEKSClustersClient) CheckAgentAlreadyInstalled(kubeconfig genericclioptions.RESTClientGetter, log logrus.FieldLogger) (bool, error) {
+	actionConfig, err := getHelmActionConfig(kubeconfig, log)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	return checkAgentAlreadyInstalled(actionConfig)
+}
+
+func (d *defaultEnrollEKSClustersClient) InstallKubeAgent(ctx context.Context, eksCluster *eksTypes.Cluster, proxyAddr string, kubeconfig genericclioptions.RESTClientGetter, log logrus.FieldLogger, req EnrollEKSClustersRequest) error {
+	actionConfig, err := getHelmActionConfig(kubeconfig, log)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return installKubeAgent(ctx, eksCluster, proxyAddr, actionConfig, getHelmSettings(), req)
+}
+
+// NewEnrollEKSClustersClient returns new client that can be used to enroll EKS clusters into Teleport.
+func NewEnrollEKSClustersClient(ctx context.Context, req *AWSClientRequest) (EnrollEKSCLusterClient, error) {
+	eksClient, err := newEKSClient(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	stsClient, err := newSTSClient(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt := defaultEnrollEKSClustersClient{
+		Client:    eksClient,
+		stsClient: stsClient,
+	}
+
+	return &clt, nil
+}
+
+// EnrollEKSClustersRequest contains the required fields to enroll EKS cluster to Teleport.
+type EnrollEKSClustersRequest struct {
+	// Region is the AWS Region.
+	Region string
+
+	// ClusterNames is name of the EKS cluster to enroll.
+	ClusterNames []string
+
+	// JoinToken is the token that will be used by the agent to join Teleport cluster.
+	JoinToken string
+
+	// ResourceID is the resource ID that will be put as a label to the enrolled cluster.
+	ResourceID string
+
+	// EnableAppDiscovery specifies if we should enable Kubernetes App Discovery inside the enrolled EKS cluster.
+	EnableAppDiscovery bool
+
+	// EnableAutoUpgrades specifies if we should enable agent auto upgrades.
+	EnableAutoUpgrades bool
+
+	// IsCloud specifies if enrollment is done for the Teleport Cloud client.
+	IsCloud bool
+}
+
+// EnrollEKSClusters enrolls an EKS clusters into Teleport by installing teleport-kube-agent chart on the clusters.
+func EnrollEKSClusters(ctx context.Context, log logrus.FieldLogger, clock clockwork.Clock, proxyAddr string, credsProvider aws.CredentialsProvider, clt EnrollEKSCLusterClient, req EnrollEKSClustersRequest) *EnrollEKSClusterResponse {
+	var mu sync.Mutex
+	var results []EnrollEKSClusterResult
+
+	group, groupCtx := errgroup.WithContext(ctx)
```

Comment:
```suggestion
	group, ctx := errgroup.WithContext(ctx)
```

replace ctx so people won't use the incorrect ctx

Reply:
Replaced 👍